server/index.js

```diff
@@ -59,6 +60,105 @@ app.get('/api/products/:productId', (req, res, next) => {
   }
 });
 
+app.get('/api/cart/', (req, res, next) => {
+  if (!req.session.cartId) {
+    res.json([]);
+  } else {
+    const cartSQL = `select "c"."cartItemId",
+                      "c"."price",
+                      "p"."productId",
+                      "p"."image",
+                      "p"."name",
+                      "p"."shortDescription"
+                      from "cartItems" as "c"
+                      join "products" as "p" using ("productId")
+                      where "c"."cartId" = $1`;
+    const params = [req.session.cartId];
+    return db.query(cartSQL, params)
+      .then(result => {
+        res.status(200).json(result.rows);
+        // console.log(result);
+      });
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = [req.body.productId];
+  if (!req.body.productId || isNaN(req.body.productId)) {
+    res.status(400).json({
+      error: 'Invalid or no input.'
+    });
+  } else {
+    const priceSQL = `select "price" from "products"
+                    where "productId"=$1`;
+    const params = productId;
+    db.query(priceSQL, params)
+      .then(result => {
+        if (result.rows.length === 0) {
+          next(new ClientError('No rows!', 400));
```

Comment:
You should `throw` this error instead of passing it to `next()` to shortcut to your `.catch()` callback